lib/adapters/websockex.ex

```diff
@@ -0,0 +1,84 @@
+if Code.ensure_loaded?(WebSockex) do
+  defmodule Janus.Transport.WS.Adapters.WebSockex do
+    @moduledoc """
+    Adapter for [WebSockex](https://github.com/Azolo/websockex).
+    """
+
+    use Janus.Transport.WS.Adapter
+    use WebSockex
+
+    @impl true
+    def connect(url, message_receiver, opts) do
+      timeout = opts[:timeout] || 5000
+
+      args = %{
+        message_receiver: message_receiver,
+        notify_on_connect: self()
+      }
+
+      case start_link(url, args) do
+        {:ok, ws} ->
+          # process have started but connection may still not be made
+          # therefore wait for response from handle_connect callback
+          receive do
+            {:connected, _connection} ->
+              {:ok, ws}
+          after
+            timeout ->
+              # ws might still try to connect, kill so it can stop
+              # websockex has no option to cancel connection (from what I've tried to find)
+              Process.exit(ws, :kill)
+              {:error, :connection_timeout}
+          end
+
+        {:error, _} = error ->
+          error
+      end
+    end
+
+    @impl true
+    def send(client, payload) do
+      if Process.info(client) do
+        WebSockex.send_frame(client, {:text, payload})
+      else
+        {:error, :connection_down}
+      end
+    end
+
+    @impl true
+    def disconnect(client) do
+      Kernel.send(client, :disconnect)
```

Comment:
```suggestion
      send(client, :disconnect)
```
I am pretty sure I've risen this topic on this repo before.

Reply:
I shouldn't have resolved that. Adapter has `send` callback which interferes with Kernel's function.